libpod/image/image.go

```diff
@@ -1075,3 +1076,75 @@ func (i *Image) Comment(ctx context.Context, manifestType string) (string, error
 	}
 	return ociv1Img.History[0].Comment, nil
 }
+
+// Tree gets dependencies of image
+func (i *Image) Tree(ctx context.Context) ([]Image, error) {
+	treeDepth := 1
+	rootImage := i
+
+	//Traverse to the root(base) image
+	parent, err := i.GetParent()
+	if err != nil {
+		return nil, err
+	}
+
+	for parent != nil {
+		rootImage = parent
+		parent, err = parent.GetParent()
+		if err != nil {
+			break
+		}
+		treeDepth++
+	}
+
+	//Print all subtree(peers) to the depth of input image
+	printSubTree(rootImage, 0, "└─ ", treeDepth)
+
+	return nil, err
+}
+
+func printSubTree(root *Image, rootSize uint64, prefix string, treeDepth int) {
+	if treeDepth == 0 {
+		return
+	}
+	size, err := root.Size(context.Background())
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	//Size of current layer will be current layer size minus upper layer
+	fmt.Printf("%s ID: %s  Tags: %v VirtualSize: %s\n", prefix, root.ID()[:12], root.Names(), humanSize((*size)-rootSize))
+
+	children, err := root.GetChildren()
+	if err != nil {
+		return
+	}
+
+	for _, child := range children {
+		img, err := root.imageruntime.getImage(child)
+		if err != nil {
+			break
+		}
+		nextPrefix := "  "
+		printSubTree(img, *size, nextPrefix+prefix, treeDepth-1)
+
+	}
+}
+
+func humanSize(raw uint64) string {
```

Comment:
We use
	"github.com/docker/go-units"

And

size = units.HumanSizeWithPrecision(0, 0)